crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -239,61 +240,52 @@ impl ObjectStore for AuthorityPerpetualTables {
             .skip_prior_to(&ObjectKey::max_for_id(object_id))?
             .next();
 
-        let obj = match obj_entry {
-            Some((ObjectKey(obj_id, _), obj)) if obj_id == *object_id => obj,
-            _ => return Ok(None),
-        };
-
-        // Note that the two reads in this function are (obviously) not atomic, and the
-        // object may be deleted after we have read it. Hence we check get_latest_parent_entry
-        // last, so that the write to self.parent_sync gets the last word.
-        // Such race is ok because, even if the reads were atomic, calls to this function would still
-        // race with object deletions (the object could be deleted between when the function is
-        // called and when the first read takes place, which would be indistinguishable to the
-        // caller with the case in which the object is deleted in between the two reads).
-        let parent_entry = self.get_latest_parent_entry(*object_id)?;
-
-        match parent_entry {
-            None => {
-                error!(
-                    ?object_id,
-                    "Object is missing parent_sync entry, data store is inconsistent"
-                );
-                Ok(None)
-            }
-            Some((obj_ref, _)) if obj_ref.2.is_alive() => Ok(Some(self.object(obj)?)),
+        match obj_entry {
+            Some((ObjectKey(obj_id, _), obj)) if obj_id == *object_id => Ok(self.object(obj)?),
             _ => Ok(None),
         }
     }
 }
 
 pub struct LiveSetIter<'a> {
-    iter: <DBMap<ObjectRef, TransactionDigest> as Map<'a, ObjectRef, TransactionDigest>>::Keys,
-    prev: Option<ObjectRef>,
+    iter:
+        <DBMap<ObjectKey, StoreObjectWrapper> as Map<'a, ObjectKey, StoreObjectWrapper>>::Iterator,
+    tables: &'a AuthorityPerpetualTables,
+    prev: Option<(ObjectKey, StoreObjectWrapper)>,
 }
 
 impl Iterator for LiveSetIter<'_> {
     type Item = ObjectRef;
 
     fn next(&mut self) -> Option<Self::Item> {
         loop {
-            if let Some(next) = self.iter.next() {
-                let prev = self.prev;
-                self.prev = Some(next);
-
-                match prev {
-                    Some(prev) if prev.0 != next.0 && prev.2.is_alive() => return Some(prev),
-                    _ => continue,
+            if let Some((next_key, next_value)) = self.iter.next() {
```

Comment:
hm, on a totally different note, I'm a little bit confused about our rocksdb iterator usage.
In all call sites we act like rocksdb iterator can't fail.
We do check if iterator is still [valid](https://github.com/MystenLabs/sui/blob/main/crates/typed-store/src/rocks/iter.rs#L45), but don't check the [status](https://docs.rs/rocksdb/latest/rocksdb/struct.DBRawIteratorWithThreadMode.html#method.status). So we can't really differentiate the end of iteration vs rocksdb error, right?

Reply:
i think that's correct. is that just a bug in our TypedStore api? if so we should probably fix it so that next() returns a result